internal/compiler/passes/embed_glyphs.rs

```diff
@@ -145,67 +163,106 @@ pub fn embed_glyphs<'a>(
+                    _ => {
+                        diag.push_error(
+                            "internal error: memory fonts are not supported in the compiler"
+                                .to_string(),
+                            &generic_diag_location,
+                        );
+                        custom_face_error = true;
+                        return None;
+                    }
                 },
                 *face_id,
-            )
+            ))
         })
     }));
 
-    let embed_font_by_path_and_face_id = |path, face_id| {
-        let font = fontdb
+    if custom_face_error {
+        return;
+    }
+
+    let mut embed_font_by_path_and_face_id = |path, face_id| {
+        let maybe_font = if let Some(maybe_font) = fontdb
             .with_face_data(face_id, |font_data, face_index| {
-                let font = fontdue::Font::from_bytes(
-        font_data,
-        fontdue::FontSettings { collection_index: face_index, scale: 40. },
-    )
-    .expect("internal error: fontdb returned a font that ttf-parser/fontdue could not parse");
+                let fontdue_font = if let Ok(fontdue_font) = fontdue::Font::from_bytes(font_data, fontdue::FontSettings { collection_index: face_index, scale: 40. }) {
+                    fontdue_font
+                } else {
+                    diag.push_error("internal error: mfontdb returned a font that ttf-parser/fontdue could not parse".to_string(), &generic_diag_location);
+                    return None;
+                };
+
+                let family_name = if let Some(family_name) = fontdb
+                    .face(face_id)
+                    .expect("must succeed as we are within face_data with same face_id")
+                    .families
+                    .first()
+                    .map(|(name, _)| name.clone()) {
+                    family_name
+                } else {
+                    diag.push_error("internal error: TrueType font without english family name encountered".to_string(), &generic_diag_location);
+                    return None;
+                };
+
                 embed_font(
-                    fontdb
-                        .face(face_id)
-                        .unwrap()
-                        .families
-                        .first()
-                        .expect("TrueType font without english family name encountered")
-                        .0
-                        .clone(),
-                    font,
+                        family_name,
+                        fontdue_font,
                     &pixel_sizes,
                     characters_seen.iter().cloned(),
                     &fallback_fonts,
-                )
-            })
-            .unwrap();
+                ).into()
+            }) {
+            maybe_font
+        } else {
+            diag.push_error("internal error: face_id of selected font is unknown to fontdb".to_string(), &generic_diag_location);
```

Comment:
(also valid for other error messages) Is there a way to specify which font is incorrect? 
As an user, if i'm getting this error, i don't really know what font went wrong.

Reply:
Hmmmn, maybe. This particular situation really can't quite happen, but for debugging purposes I can try to enhance this and others.